cas-server-support-saml/src/main/java/org/jasig/cas/support/saml/web/flow/mdui/SamlMetadataUIParserAction.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Licensed to Apereo under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work
+ * for additional information regarding copyright ownership.
+ * Apereo licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License.  You may obtain a
+ * copy of the License at the following location:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.jasig.cas.support.saml.web.flow.mdui;
+
+import org.apache.commons.lang3.StringUtils;
+import org.jasig.cas.authentication.principal.SimpleWebApplicationServiceImpl;
+import org.jasig.cas.authentication.principal.WebApplicationService;
+import org.jasig.cas.services.RegisteredService;
+import org.jasig.cas.services.ServicesManager;
+import org.jasig.cas.services.UnauthorizedServiceException;
+import org.jasig.cas.web.support.WebUtils;
+import org.opensaml.core.xml.XMLObject;
+import org.opensaml.saml.common.xml.SAMLConstants;
+import org.opensaml.saml.ext.saml2mdui.UIInfo;
+import org.opensaml.saml.saml2.metadata.EntityDescriptor;
+import org.opensaml.saml.saml2.metadata.Extensions;
+import org.opensaml.saml.saml2.metadata.SPSSODescriptor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.webflow.action.AbstractAction;
+import org.springframework.webflow.execution.Event;
+import org.springframework.webflow.execution.RequestContext;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.validation.constraints.NotNull;
+import java.util.List;
+
+/**
+ * This is {@link SamlMetadataUIParserAction} that attempts to parse
+ * the mdui extension block for a SAML SP from the provided metadata locations.
+ * The result is put into the flow request context under the parameter
+ * {@link #MDUI_FLOW_PARAMETER_NAME}. The entity id parameter is
+ * specified by default at {@link #ENTITY_ID_PARAMETER_NAME}.
+ *
+ * <p>This action is best suited to be invoked when the CAS login page
+ * is about to render so that the page, once the MDUI info is obtained,
+ * has a chance to populate the UI with relevant info about the SP.</p>
+ *
+ * @author Misagh Moayyed
+ * @since 4.1.0
+ */
+public class SamlMetadataUIParserAction extends AbstractAction {
```

Comment:
No because argument extractors attempts to create services, or specific service types which we don't want to do here. We simply want to be able to manipulate the flow when it starts up.